web/packages/teleterm/src/ui/Search/useSearch.ts

```diff
@@ -360,6 +364,11 @@ function calculateScore(
     searchResultScore += resourceMatchScore;
   }
 
+  // Show resources that require access at the bottom.
+  if (searchResult.requiresRequest) {
+    searchResultScore *= 0.1;
```

Comment:
This doesn't prevent the normal resource from having a score lower than a requestable one, but perhaps it is good enough?

Reply:
What's the reasoning behind such a severe penalty though? I think something like `*= 0.95` should be good enough for now.

Imagine this scenario:

```diff
diff --git a/web/packages/teleterm/src/ui/Search/useSearch.test.tsx b/web/packages/teleterm/src/ui/Search/useSearch.test.tsx
index 5d7e76d6ff..2f87de779b 100644
--- a/web/packages/teleterm/src/ui/Search/useSearch.test.tsx
+++ b/web/packages/teleterm/src/ui/Search/useSearch.test.tsx
@@ -54,6 +54,33 @@ describe('rankResults', () => {
     expect(sortedResults[1]).toEqual(server);
   });
 
+  it('prefers accessible resources over requestable ones', () => {
+    const serverAccessible = makeResourceResult({
+      kind: 'server',
+      resource: makeServer({ hostname: 'sales-foo' }),
+    });
+    const serverRequestable = makeResourceResult({
+      kind: 'server',
+      resource: makeServer({ hostname: 'sales-bar' }),
+      requiresRequest: true,
+    });
+    const labelMatch = makeResourceResult({
+      kind: 'server',
+      resource: makeServer({
+        hostname: 'lorem-ipsum',
+        labels: makeLabelsList({ foo: 'sales' }),
+      }),
+    });
+    const sortedResults = rankResults(
+      [labelMatch, serverRequestable, serverAccessible],
+      'sales'
+    );
+
+    expect(sortedResults[0].resource).toEqual(serverAccessible.resource);
+    expect(sortedResults[1].resource).toEqual(serverRequestable.resource);
+    expect(sortedResults[2].resource).toEqual(labelMatch.resource);
+  });
+
   it('saves individual label match scores', () => {
     const server = makeResourceResult({
       kind: 'server',

```

The scores are as follows:

```
yours (=* 0.1):
[ 'sales-foo: 220', 'lorem-ipsum: 100', 'sales-bar: 22' ]

mine (=* 0.95):
[ 'sales-foo: 220', 'sales-bar: 209', 'lorem-ipsum: 100' ]
```

In this scenario, I don't believe we'd want a resource with a single hit on a label to be above a requestable resource with a hit on its name.

Reply:
Ok, I could be too aggressive on that 😅 My first instinct was that requestable resources should be always displayed at the bottom, but yeah, just having a lower priority may be fine.

Changed to 0.95, I also added the test, thanks!